src/Type/Row/Homogeneous.purs

```diff
@@ -0,0 +1,22 @@
+module Type.Row.Homogeneous
+  ( class FieldOf
+  , class Homogeneous
+  ) where
+
+-- | Ensure that every field in a row has the same type.
+import Type.Equality (class TypeEquals)
+import Type.Row (class RowToList, Cons, Nil, kind RowList)
+
+-- | Ensure that every field in a row has the same type.
+class Homogeneous (row :: # Type) fieldType | row -> fieldType
+instance homogeneous
+  :: ( RowToList row fields
+     , FieldOf fields fieldType )
+  => Homogeneous row fieldType
+
+class FieldOf (rowList :: RowList) fieldType | rowList -> fieldType
+instance fieldOfCons
+  :: ( FieldOf tail fieldType
+     , TypeEquals fieldType fieldType2 )
```

Comment:
Does this need `TypeEquals`? Can you just use 

```purescript
instance fieldOfCons :: FieldOf tail fieldType => FieldOf (Cons symbol fieldType tail) fieldType
```

Reply:
I did that in https://github.com/purescript/purescript-typelevel-prelude/pull/20/commits/8b39b00eab1e640a0fe3af7badcf3af0218240c3#diff-86491d4ab8759f2c9d61332f39afdd9bR15 but you suggested using `TypeEquals` in https://github.com/purescript/purescript-typelevel-prelude/pull/20#issuecomment-321603793 😄

Reply:
D'oh. Thanks :)